src/main/java/io/jenkins/plugins/opentelemetry/servlet/StaplerInstrumentationServletFilter.java

```diff
@@ -87,174 +96,173 @@ public void doFilter(ServletRequest servletRequest, ServletResponse servletRespo
-                    httpRoute = "/blue/";
-                    spanBuilder = tracer.spanBuilder(servletRequest.getMethod() + " " + httpRoute);
-                } else if ("rest".equals(pathInfoTokens.get(1))) {
-                    if (pathInfoTokens.size() == 2) {
-                        httpRoute = "/blue/rest/";
-                        spanBuilder = tracer.spanBuilder(servletRequest.getMethod() + " " + httpRoute);
-                    } else if ("organizations".equals(pathInfoTokens.get(2)) && pathInfoTokens.size() > 7) {
-                        // eg /blue/rest/organizations/jenkins/pipelines/ecommerce-antifraud/branches/main/runs/110/blueTestSummary/
-
-                        ParsedJobUrl parsedBlueOceanPipelineUrl = parseBlueOceanRestPipelineUrl(pathInfoTokens);
-                        httpRoute = parsedBlueOceanPipelineUrl.urlPattern;
-                        spanBuilder = tracer.spanBuilder(servletRequest.getMethod() + " " + parsedBlueOceanPipelineUrl.urlPattern);
-                        if (parsedBlueOceanPipelineUrl.jobName != null) {
-                            spanBuilder.setAttribute(JenkinsOtelSemanticAttributes.CI_PIPELINE_ID, parsedBlueOceanPipelineUrl.jobName);
+
+            List<String> pathInfoTokens = Collections.list(new StringTokenizer(servletRequest.getPathInfo(), "/")).stream()
+                .map(token -> (String) token)
+                .filter(t -> !t.isEmpty())
+                .collect(Collectors.toList());
+
+            if (pathInfoTokens.isEmpty()) {
+                pathInfoTokens = Collections.singletonList("");
+            }
+
+            String rootPath = pathInfoTokens.get(0);
+            String httpRoute;
+
+            boolean skipSpan = false;
+            try {
+                switch (rootPath) {
+                    case "job" -> {
+                        // e.g /job/my-war/job/master/lastBuild/console
+                        // e.g /job/my-war/job/master/2/console
+                        ParsedJobUrl parsedJobUrl = parseJobUrl(pathInfoTokens);
+                        httpRoute = parsedJobUrl.urlPattern;
+                        Optional.ofNullable(parsedJobUrl.jobName).ifPresent(jobName -> httpServerSpanAttributesBuilder.put(JenkinsOtelSemanticAttributes.CI_PIPELINE_ID, jobName));
+                        Optional.ofNullable(parsedJobUrl.runNumber).ifPresent(runNumber -> httpServerSpanAttributesBuilder.put(JenkinsOtelSemanticAttributes.CI_PIPELINE_RUN_NUMBER, runNumber));
+                    }
+                    case "blue" -> {
+                        if (pathInfoTokens.size() == 1) {
+                            httpRoute = "/blue/";
+                        } else if ("rest".equals(pathInfoTokens.get(1))) {
+                            if (pathInfoTokens.size() == 2) {
+                                httpRoute = "/blue/rest/";
+                            } else if ("organizations".equals(pathInfoTokens.get(2)) && pathInfoTokens.size() > 7) {
```

Comment:
5 complexity levels

Reply:
I know, it's why it's full of unit tests :-)
Using Java 17 `switch` syntax is already an improvement over the `if .. else if ...` we used to have.

I think that the durable solution would be to instrument the Stapler framework itself but I don't know how to make this happen.

Reply:
I will propose a refactor in other PR